First iteration of refactoring

Ziticca opened the pull request.

hj799web commented:
Great job? Just a few questions

Did you test all major functionalities after refactoring? Just to confirm that no regressions were introduced

hj799web commented:
Validation (86e91d9): Great to see validation added! Are there user-friendly error messages when validation fails?

Ziticca commented:
> Great job? Just a few questions
> 
> Did you test all major functionalities after refactoring? Just to confirm that no regressions were introduced

I did that all calculations work, charts are displayed and the budget categories work as they did previously. I did not change any calculation formulas, so they should work like previously, but I'd like you to double check them.

Ziticca commented:
> Validation ([86e91d9](https://github.com/hj799web/Tax-calculator-website/commit/86e91d99111847d7085e55bbdfdaa4ad5cd2af59)): Great to see validation added! Are there user-friendly error messages when validation fails?

Right now there's rather generic "Value must be a number greater than 0" in all the number fields, but it can be customized to whatever is wanted

hj799web reviewed (changes requested):
After testing the changes in this pull request, I noticed two major issues:
1. The tax calculations are significantly off. The results are not aligning with expected values, especially for federal and provincial taxes. Please review the calculation logic and constants used in the `useCalculatorStore` file.
2. The placeholder text in input fields is no longer italicized. This was part of the original design and should be restored.

Let me know if you need further details or examples to reproduce these issues.

hj799web commented:
The size of the pie charts are perfect!

Ziticca commented:
> After testing the changes in this pull request, I noticed two major issues:
> 
>     1. The tax calculations are significantly off. The results are not aligning with expected values, especially for federal and provincial taxes. Please review the calculation logic and constants used in the `useCalculatorStore` file.
> 
>     2. The placeholder text in input fields is no longer italicized. This was part of the original design and should be restored.
> 
> 
> Let me know if you need further details or examples to reproduce these issues.

These should be now fixed. There was an issue with handling the numerical values as texts as I had changed the input fields to be text fields instead of numbers to help with validation. I've now changed them back to number fields and tuned the validation so that it should work with number fields

hj799web commented:
> > After testing the changes in this pull request, I noticed two major issues:
> > ```
> > 1. The tax calculations are significantly off. The results are not aligning with expected values, especially for federal and provincial taxes. Please review the calculation logic and constants used in the `useCalculatorStore` file.
> > 
> > 2. The placeholder text in input fields is no longer italicized. This was part of the original design and should be restored.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Let me know if you need further details or examples to reproduce these issues.
> 
> These should be now fixed. There was an issue with handling the numerical values as texts as I had changed the input fields to be text fields instead of numbers to help with validation. I've now changed them back to number fields and tuned the validation so that it should work with number fields
